pkgs/tools/typesetting/tex/texlive/combine.nix

```diff
@@ -122,22 +122,21 @@ in (buildEnv {
   '' +
     # now filter hyphenation patterns, in a hacky way ATM
   (let
-    pnames = uniqueStrings (map (p: p.pname) pkgList.splitBin.wrong);
-    script =
-      writeText "hyphens.sed" (
+    hyphens = lib.filter (p: p.hasHyphens or false) pkgList.splitBin.wrong;
+    pnames = uniqueStrings (map (p: p.pname) hyphens);
+    sedExprs =
         # pick up the header
-        "1,/^% from/p;"
+        "-e '1,/^% from/p;' "
         # pick up all sections matching packages that we combine
-        + lib.concatMapStrings (pname: "/^% from ${pname}:$/,/^%/p;\n") pnames
-      );
+        + lib.concatMapStrings (pname: "-e '/^% from ${pname}:$/,/^%/p;' ") pnames;
   in ''
     (
       cd ./share/texmf/tex/generic/config/
       for fname in language.dat language.def; do
         [ -e $fname ] || continue;
         cnfOrig="$(realpath ./$fname)"
         rm ./$fname
-        cat "$cnfOrig" | sed -n -f '${script}' > ./$fname
+        cat "$cnfOrig" | sed -n ${sedExprs} > ./$fname
```

Comment:
Files are better than than arguments because they don't have size limits

Reply:
I will change it to `passAsFile`.

Reply:
I have got an idea. Why don't we make the output files `language.{dat,def,def.lua}` into their own derivation and we pass them to `buildEnv` together with `pkgList.nonbin`? ~It sounds cleaner.~ Ugh, we would also need to set high priority to prevent the conflict with `kpathsea`. So not *that* cleaner.

Reply:
...and `buildEnv` does not take `passAsFile` as argument. So we *must* use an extra derivation, containing either `hyphens{.lua}.sed` or the actual config files. I am partial to the latter (and `runCommand` does support `passAsFile`, so no issues there). In fact, it even generalises better if we also create `fmtutil.cnf`.

Reply:
You don't need to invent anything new here, it is already writing a file right now

Reply:
I rewrote the entire PR to use `hyphens{.lua}.sed`. It's a smaller change now.

Reply:
Great